gtfu/src/main/java/gtfu/GraphicReport.java

```diff
@@ -195,17 +205,24 @@ public GraphicReport(String cacheDir, String selectedDate, String savePath, bool
         return out.toByteArray();
     }
 
-    private void sendEmail(List<byte[]> blobs) throws IOException {
+    private void sendEmail(Map<String, byte[]> blobMap) throws IOException {
         Debug.log("GraphicReport.sendEmail()");
         // Debug.log("- blobs: " + blobs);
 
         Recipients r = new Recipients();
         String[] recipients = r.get("graas_report");
 
-        SendGrid grid = new SendGrid(recipients, "Automated GRaaS Report", "Attached", blobs);
+        SendGrid grid = new SendGrid(recipients, "Automated GRaaS Report", "Attached", blobMap);
         int responseCode = grid.send();
     }
 
+    private void uploadToGCloud(String key, byte[] image) throws IOException {
+        // converts <agency-id>-yyyy-mm-dd to <agency-id>
+        String agencyID = key.substring(0, key.length() - 11);
+        String path = "graas-report-archive/" + agencyID;
+        gcs.uploadObject("graas-resources", path, key, image);
```

Comment:
if key is `<agency-id>-yyyy-mm-dd` where does the file name suffix, i.e. `png` come from?